### PR TITLE
add Note Feature

### PR DESCRIPTION
**Pull request by atharvabhanage02**
Features added in this PR

- Create Note
- Add color to the Note (the colors added are demo colors and will be changed soon)
- Set Priority (Low/Medium/High)
- Update Note

Files to look for:

- notesContext : `src/Context/NotesContext/NotesContext.js`
- Services folder : `src/Services`
- Reducer : `src/Reducer/noteReducer.js`
- Components : `src/Components` (ColorPallete , Priority , Notes , NotesCard , NotesDisplay )
- Page : `src/Pages/NotesPage.jsx`

### CODE REVIEW

File: src/Context/NotesContext/NotesContext.js [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+  useEffect,
+  useReducer,
+  useState,
+} from "react";
+import { useAuth } from "../Auth/auth-context";
+import { initialNote, noteReducer } from "../../Reducers/noteReducer";
+import { getNotes } from "../../Services/getNotes";
+import { addNoteService } from "../../Services/addNote";
+import { updateNotesService } from "../../Services/updateNote";
+import { deleteNoteService } from "../../Services/deleteNote";
```
Comment (maitrakhatri) [inline src/Context/NotesContext/NotesContext.js:13]: use barrel exports

File: src/Pages/Notes/NotesPage.jsx [hunk 2]
```diff
@@ -1,11 +1,13 @@
 import { Navbar } from "../../components/Navbar/Navbar";
+import { Notes } from "../../components/Notes/Notes";
 import { Sidebar } from "../../components/Sidebar/Sidebar";
```
Comment (maitrakhatri) [inline src/Pages/Notes/NotesPage.jsx:3]: barrel exports

File: src/components/Notes-Card/notescard.css [hunk 3]
```diff
@@ -0,0 +1,130 @@
+:root {
+  --notes-red-bg: red;
+  --notes-blue-bg: blue;
+  --notes-green-bg: green;
+  --notes-yellow-bg: yellow;
```
Comment (maitrakhatri) [inline src/components/Notes-Card/notescard.css]: use hex codes, as names differs from browser to browser. Your app will not look consistent throughout different browsers

**Review by maitrakhatri — APPROVED** [review]
Code is clean, easy to read.
Left few suggestion, rest looks good to me.,

File: src/components/NotesTemplate/NotesDisplay.jsx [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+        )}
+        {pathname === "/notes" && (
+          <BiEdit
+            className="notes-card-icons"
+            onClick={() =>
+              dispatchNote({ type: "UPDATE_NOTE", payload: notesData })
+            }
+          />
+        )}
+      </div>
```
Comment (dev-enforced) [inline src/components/NotesTemplate/NotesDisplay.jsx:56]: This is a new logic for me of including options!! Clever step @atharvabhanage02 😃

File: src/components/NotesTemplate/NotesDisplay.jsx [hunk 5]
```diff
@@ -0,0 +1,55 @@
+import "./notes-display.css";
+import { BiArchiveIn, BiEdit, BiArchiveOut } from "react-icons/bi";
+import { FaTrashRestoreAlt } from "react-icons/fa";
+import { BsPin, BsPinFill } from "react-icons/bs";
+import { AiOutlineDelete } from "react-icons/ai";
```
Comment (dev-enforced) [inline src/components/NotesTemplate/NotesDisplay.jsx]: Would be better to have a separate files for icons making it easy for you to import from a single line multiple items.

File: src/Reducers/noteReducer.js [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+const noteReducer = (state, action) => {
+  switch (action.type) {
+    case "SET_TITLE":
+      return { ...state, title: action.payload };
+    case "SET_DESCRIPTION":
+      return { ...state, description: action.payload };
+    case "SET_COLOR":
+      return { ...state, color: action.payload };
+    case "SET_PRIORITY":
+      console.log("Code reached reducer of Priority");
```
Comment (dev-enforced) [inline src/Reducers/noteReducer.js]: Can remove this debugger!!

**Review by dev-enforced — APPROVED** [review]
The code looks really very well structured. Do apply the suggestions given by @maitrakhatri very thoughtful they are!!
The live preview also looks good loved the animations when placing my cursor over the buttons of the notes.  https://deploy-preview-2--eclectic-choux-d9c480.netlify.app/notes
Good work @atharvabhanage02

File: src/components/NotesTemplate/NotesDisplay.jsx (hunk 5 above)
Comment (atharvabhanage02) [reply src/components/NotesTemplate/NotesDisplay.jsx]: Done 👍

**Pull request merged**